src/dshould/basic.d

```diff
@@ -0,0 +1,328 @@
+module dshould.basic;
+
+import std.string : empty;
+import dshould.ShouldType;
+public import dshould.ShouldType : should;
+
+pure @safe unittest
+{
+    2.should.be(2);
+    2.should.not.be(5);
+    5.should.equal(5);
+    5.should.not.equal(6);
+    5.should.not.be.equal.greater(6);
+    5.should.be.equal.greater(5);
+    5.should.be.equal.greater(4);
+    5.should.not.be.equal.greater(6);
+    5.should.be.smaller(6);
+}
+
+unittest
+{
+    (new Object).should.not.be(new Object);
+    (new Object).should.not.equal(new Object);
+    (new Object).should.not.be(null);
+    (cast(Object) null).should.be(null);
+
+    auto obj = new Object;
+
+    obj.should.equal(obj);
+    obj.should.be(obj);
+
+    class SameyClass
+    {
+        override bool opEquals(Object o) { return true; }
+    }
+
+    (new SameyClass).should.not.be(new SameyClass);
+    (new SameyClass).should.equal(new SameyClass);
+
+    (cast(void delegate()) null).should.be(null);
+}
+
+auto not(Should)(Should should) pure
+if (isInstanceOf!(ShouldType, Should))
+{
+    should.allowOnlyWordsBefore!([], "not");
```

Comment:
if you initialize `words` with `["should"]`, this would be more explicit:
"not" has to follow "should" immediately

Reply:
that would mean I have to redundantly add "should" in every call to allowOnlyWordsBefore...